### PR TITLE
Setting up linting and testing

### PR DESCRIPTION
**Pull request by fredericpoitevin**
This is our first attempt at setting up github workflows for linting and testing. 

We also added unit tests after editing the code so it follows our contributing guidelines more closely.

### CODE REVIEW

File: iospi/iotools/atomic_models.py
Comment (ninamiolane) [inline iospi/iotools/atomic_models.py:18]: NIT: I don't think we need the parentheses for the types of the parameters.

File: .github/workflows/build.yml
Comment (ninamiolane) [inline .github/workflows/build.yml:26]: We might want to drop python 3.6: I don't know how much it's used now?

File: .github/workflows/build.yml
Comment (ninamiolane) [inline .github/workflows/build.yml:42]: let's install flake8-black right away, instead of flake8 (different from what we do in geomstats): flake8 and black have different conventions: we might get weird errors if we don't use flake8-black:
https://pypi.org/project/flake8-black/

**Review by ninamiolane — CHANGES_REQUESTED** [review]
That looks great, thank you so much!

One important comment to address before merging: let's use flake8-black instead of flake8, otherwise we will have conflicts with the deepsource bot using black.

Next: we should add some type of code coverage, to have a sense of how much of the library is being unittested (hopefully, > 90%).

Thank you so much for doing this!

File: iospi/iotools/atomic_models.py
Comment (fredericpoitevin) [reply iospi/iotools/atomic_models.py]: Thanks! That's true looks better without them

File: .github/workflows/build.yml
Comment (fredericpoitevin) [reply .github/workflows/build.yml]: Good point, removed it.

File: .github/workflows/build.yml
Comment (fredericpoitevin) [reply .github/workflows/build.yml]: Yes makes sense. I replaced, let's see how it works.

**Comment by fredericpoitevin** [comment]
> Next: we should add some type of code coverage, to have a sense of how much of the library is being unittested (hopefully, > 90%).

Yes, I quote reply to this to make sure it does not go under the rug. I want to do this asap.

**Comment by ninamiolane** [comment]
I see this error:
```
ERROR: test_atomic_models (nose2.loader.ModuleImportFailure)
----------------------------------------------------------------------
ImportError: Failed to import test module: test_atomic_models
Traceback (most recent call last):
  File "/usr/share/miniconda/lib/python3.8/site-packages/nose2/plugins/loader/discovery.py", line 201, in _find_tests_in_file
    module = util.module_from_name(module_name)
  File "/usr/share/miniconda/lib/python3.8/site-packages/nose2/util.py", line 77, in module_from_name
    __import__(name)
  File "/home/runner/work/ioSPI/ioSPI/tests/tests_iotools/test_atomic_models.py", line 6, in <module>
    from iotools.atomic_models import read_gemmi_model, write_gemmi_model
ModuleNotFoundError: No module named 'iotools'
```
Maybe `from iospi.iotools.atomic_models import read_gemmi_model, write_gemmi_model`?

**Comment by ninamiolane** [comment]
@fredericpoitevin Sorry to be a pain, but I actually think that we should use `pytest` instead of `nose2` to run the unit-tests:
https://python.libhunt.com/compare-pytest-vs-nose2

Geomstats has been using `nose2`, but we are moving to `pytest`.

**Comment by fredericpoitevin** [comment]
> I see this error:
> 
> ```
> ERROR: test_atomic_models (nose2.loader.ModuleImportFailure)
> ----------------------------------------------------------------------
> ImportError: Failed to import test module: test_atomic_models
> Traceback (most recent call last):
>   File "/usr/share/miniconda/lib/python3.8/site-packages/nose2/plugins/loader/discovery.py", line 201, in _find_tests_in_file
>     module = util.module_from_name(module_name)
>   File "/usr/share/miniconda/lib/python3.8/site-packages/nose2/util.py", line 77, in module_from_name
>     __import__(name)
>   File "/home/runner/work/ioSPI/ioSPI/tests/tests_iotools/test_atomic_models.py", line 6, in <module>
>     from iotools.atomic_models import read_gemmi_model, write_gemmi_model
> ModuleNotFoundError: No module named 'iotools'
> ```
> 
> Maybe `from iospi.iotools.atomic_models import read_gemmi_model, write_gemmi_model`?

Thanks! I finally found some time to fix it :D

**Comment by fredericpoitevin** [comment]
> @fredericpoitevin Sorry to be a pain, but I actually think that we should use `pytest` instead of `nose2` to run the unit-tests:
> https://python.libhunt.com/compare-pytest-vs-nose2
> 
> Geomstats has been using `nose2`, but we are moving to `pytest`.

Sounds good! I switched to pytest. Let's see if it goes through this time!

File: iospi.yml
Comment (ninamiolane) [inline iospi.yml:6]: We should not have any leftover comments in the main codebase. If you want to keep them somewhere, you can put them on a separate branche.

File: iospi/__init__.py
Comment (ninamiolane) [inline iospi/__init__.py:3]: No leftover comment

File: iospi/iotools/atomic_models.py
Comment (ninamiolane) [inline iospi/iotools/atomic_models.py:46]: if -> elif?

File: iospi/iotools/atomic_models.py
Comment (ninamiolane) [inline iospi/iotools/atomic_models.py:48]: use else?

File: tests/tests_iotools/test_atomic_models.py
Comment (ninamiolane) [inline tests/tests_iotools/test_atomic_models.py:7]: no leftover comment

File: tests/tests_iotools/test_atomic_models.py
Comment (ninamiolane) [inline tests/tests_iotools/test_atomic_models.py:16]: DITTO

File: tests/tests_iotools/test_atomic_models.py
Comment (ninamiolane) [inline tests/tests_iotools/test_atomic_models.py:37]: why pass here?

File: .github/workflows/build.yml
Comment (ninamiolane) [inline .github/workflows/build.yml:47]: ioSPI is sometimes called iospi and someting ioSPI: do we want to uniformize this? it could lead to weird (human or computer) errors...

File: iospi/iotools/atomic_models.py
Comment (ninamiolane) [inline iospi/iotools/atomic_models.py:1]: NIT: Write -> write

File: iospi/iotools/atomic_models.py
Comment (ninamiolane) [inline iospi/iotools/atomic_models.py:17]: MEGA NIT: I think the exact syntax has a blank space before :, but... don't know if that really matters

File: iospi/iotools/atomic_models.py
Comment (ninamiolane) [inline iospi/iotools/atomic_models.py:33]: Don't put "TODO" in the docstring, otherwise it will show up in our documentation website later - and will look unprofessional to the visitors. 
see here:
https://geomstats.github.io/geometry.html#module-geomstats.geometry.special_euclidean

You can put "TODO" as a comment in the code if you wish, or just delete it.

**Review by ninamiolane — APPROVED** [review]
Very nice! Thank you for doing this and for your amazing patience with the continuous integration. Good to merge for me when the leftover comments are deleted (+ mini NIT comments).

**Review by ninamiolane — COMMENTED** [review]
Another mini-comment (but it might save us random headaches in the future):
let's do: `tests/test_ioSPI/test_*` i.e. only a `s` for the global repo, and no `s` in the subrepos. Otherwise we can't remember which ones have an s and which one do not...

File: iospi/iotools/atomic_models.py
Comment (fredericpoitevin) [reply iospi/iotools/atomic_models.py:46]: After many iterations to satisfy the bots I ended up with this but I agree it looks weird. I'll try elif again

File: iospi/iotools/atomic_models.py
Comment (fredericpoitevin) [reply iospi/iotools/atomic_models.py:48]: Ditto

File: .github/workflows/build.yml
Comment (fredericpoitevin) [reply .github/workflows/build.yml:47]: hmmm right this could be annoying going forward... PEP8 naming conventions require naming modules with all-lowercases (https://www.python.org/dev/peps/pep-0008/#package-and-module-names). I think it is a good idea, we will need to rename all the repos like that: ioSPI ==> iospi, reduceSPI ==> reducespi, etc. 
I am going to make sure all the other comments are adressed before I try to rename the repo.

File: iospi/iotools/atomic_models.py
Comment (fredericpoitevin) [reply iospi/iotools/atomic_models.py:33]: I deleted it for now

File: iospi/iotools/atomic_models.py
Comment (fredericpoitevin) [reply iospi/iotools/atomic_models.py:17]: done

File: iospi/iotools/atomic_models.py
Comment (fredericpoitevin) [reply iospi/iotools/atomic_models.py:1]: done

File: tests/tests_iotools/test_atomic_models.py
Comment (fredericpoitevin) [reply tests/tests_iotools/test_atomic_models.py:37]: This test creates a file with the wrong extension (.txt instead of .pdb or .cif) and tests that the error gets caught. The file can be empty hence the pass - I edited to something maybe more elegant?

File: tests/tests_iotools/test_atomic_models.py
Comment (fredericpoitevin) [reply tests/tests_iotools/test_atomic_models.py:16]: done

File: tests/tests_iotools/test_atomic_models.py
Comment (fredericpoitevin) [reply tests/tests_iotools/test_atomic_models.py:7]: done

File: iospi/__init__.py
Comment (fredericpoitevin) [reply iospi/__init__.py:3]: done

File: iospi.yml
Comment (fredericpoitevin) [reply iospi.yml:6]: done

File: .github/workflows/build.yml
Comment (fredericpoitevin) [reply .github/workflows/build.yml:55]: thanks for going through all those! They should be gone now

File: .github/workflows/build.yml
Comment (fredericpoitevin) [reply .github/workflows/build.yml:47]: I think this could be addressed in another PR. I created an issue for now: https://github.com/compSPI/ioSPI/issues/5

**Pull request merged**
